test/e2e/serial/workload_placement_test.go

```diff
@@ -0,0 +1,151 @@
+
+	nropv1alpha1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1alpha1"
+	e2efixture "github.com/openshift-kni/numaresources-operator/test/utils/fixture"
+	e2enrt "github.com/openshift-kni/numaresources-operator/test/utils/noderesourcetopologies"
+	"github.com/openshift-kni/numaresources-operator/test/utils/nrosched"
+	"github.com/openshift-kni/numaresources-operator/test/utils/objects"
+	e2ewait "github.com/openshift-kni/numaresources-operator/test/utils/objects/wait"
+)
+
+var _ = Describe("[serial][disruptive][scheduler] workload placement", func() {
+	var fxt *e2efixture.Fixture
+	var nroSchedObj *nropv1alpha1.NUMAResourcesScheduler
+	var schedulerName string
+	var nrtList nrtv1alpha1.NodeResourceTopologyList
+	var nrts []nrtv1alpha1.NodeResourceTopology
+
+	BeforeEach(func() {
+		var err error
+		fxt, err = e2efixture.Setup("e2e-test-workload-placement")
+		Expect(err).ToNot(HaveOccurred(), "unable to setup test fixture")
+
+		nroSchedObj = &nropv1alpha1.NUMAResourcesScheduler{}
+		err = fxt.Client.Get(context.TODO(), client.ObjectKey{Name: nrosched.NROSchedObjectName}, nroSchedObj)
+		Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", nrosched.NROSchedObjectName)
+
+		schedulerName = nroSchedObj.Status.SchedulerName
+		Expect(schedulerName).ToNot(BeEmpty(), "cannot autodetect the TAS scheduler name from the cluster")
+
+		err = fxt.Client.List(context.TODO(), &nrtList)
+		Expect(err).ToNot(HaveOccurred())
+
+		tmPolicy := nrtv1alpha1.SingleNUMANodeContainerLevel
+		nrts = e2enrt.FilterTopologyManagerPolicy(nrtList.Items, tmPolicy)
+		if len(nrts) == 0 {
+			Skip(fmt.Sprintf("cannot find NRT with policy %q", string(tmPolicy)))
+		}
+
+		// Note that this test, being part of "serial", expects NO OTHER POD being scheduled
+		// in between, so we consider this information current and valid when the It()s run.
+	})
+
+	AfterEach(func() {
+		err := e2efixture.Teardown(fxt)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	Context("cluster with multiple worker nodes suitable", func() {
+		It("[test_id:47575] should make a pod land on a node with enough resources on a specific NUMA zone", func() {
+			requiredRes := corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("3"),
+				corev1.ResourceMemory: resource.MustParse("384Mi"),
+			}
+			pod := objects.NewTestPodPause(fxt.Namespace.Name, "testpod")
+			pod.Spec.SchedulerName = schedulerName
+			pod.Spec.Containers[0].Resources.Limits = requiredRes
+
+			By("checking the cluster can run the test workload")
+			testNRTs := e2enrt.FilterAnyZoneMatchingResources(nrts, requiredRes)
+			if len(testNRTs) < 2 {
+				Skip(fmt.Sprintf("not enough nodes which can run pod (%#v) on a single NUMA zone: found %d", requiredRes, len(testNRTs)))
+			}
+
+			By("finding the candidate node list")
+			candidateNodeNames := e2enrt.AccumulateNames(testNRTs)
+			candidateNodeNamesList := candidateNodeNames.List()
+
+			By("running the test pod")
```

Comment:
how/where did we handle setting up the cluster's NRT before scheduling the pod?

Reply:
we don't atm. We trust what RTE reports verbatim, and in this specific case we expect empty nodes (save for some reserved resources)

Reply:
in that case, the cluster does not comply with the setup of the test that "multiple nodes have the requested resources but only one node has them all on a single-numa".
IIUC this test checks if a pod was successfully placed on a node on a clean cluster (no other pods are scheduled)

Reply:
> in that case, the cluster does not comply with the setup of the test that "multiple nodes have the requested resources but only one node has them all on a single-numa".

We need to have more granular control over the cluster resources, so we'll be able to handle complex prerequisites like that. https://github.com/openshift-kni/numaresources-operator/pull/116 aims to help with that.

Reply:
on top of that, this PR of mine needs more work. Will get to that.